editor/svgutils.js

```diff
@@ -529,19 +529,387 @@ svgedit.utilities.getBBox = function(elem) {
+		var l = seg[0], pts = seg[1];
+		d += l;
+		for (i = 0; i < pts.length; i+=2) {
+			d += (pts[i] +','+pts[i+1]) + ' ';
+		}
+	});
+
+	return d
+}
+
+svgedit.utilities.getPathDFromElement = function( elem) {
+
+	// Possibly the cubed root of 6, but 1.81 works best
+	var num = 1.81;
+	var d, a, rx, ry;
+	switch (elem.tagName) {
+		case 'ellipse':
+		case 'circle':
+			a = $(elem).attr(['rx', 'ry', 'cx', 'cy']);
+			var cx = a.cx, cy = a.cy;
+			rx = a.rx;
+			ry = a.ry;
+			if (elem.tagName == 'circle') {
+				rx = ry = $(elem).attr('r');
+			}
+
+			d = svgedit.utilities.getPathDFromSegments([
+				['M',[(cx-rx),(cy)]],
+				['C',[(cx-rx),(cy-ry/num), (cx-rx/num),(cy-ry), (cx),(cy-ry)]],
+				['C',[(cx+rx/num),(cy-ry), (cx+rx),(cy-ry/num), (cx+rx),(cy)]],
+				['C',[(cx+rx),(cy+ry/num), (cx+rx/num),(cy+ry), (cx),(cy+ry)]],
+				['C',[(cx-rx/num),(cy+ry), (cx-rx),(cy+ry/num), (cx-rx),(cy)]],
+				['Z',[]]
+			]);
+			break;
+		case 'path':
+			d = elem.getAttribute('d');
+			break;
+		case 'line':
+			a = $(elem).attr(['x1', 'y1', 'x2', 'y2']);
+			d = 'M'+a.x1+','+a.y1+'L'+a.x2+','+a.y2;
+			break;
+		case 'polyline':
+			d = 'M' + elem.getAttribute('points');
+			break;
+		case 'polygon':
+			d = 'M' + elem.getAttribute('points') + ' Z';
+			break;
+		case 'rect':
+			var r = $(elem).attr(['rx', 'ry']);
+			rx = r.rx;
+			ry = r.ry;
+			var b = elem.getBBox();
+			var x = b.x, y = b.y, w = b.width, h = b.height;
+			num = 4 - num; // Why? Because!
+
+			if (!rx && !ry) {
+				// Regular rect
+				d = svgedit.utilities.getPathDFromSegments([
+					['M',[x, y]],
+					['L',[x+w, y]],
+					['L',[x+w, y+h]],
+					['L',[x, y+h]],
+					['L',[x, y]],
+					['Z',[]]
+				]);
+			} else {
+				d = svgedit.utilities.getPathDFromSegments([
+					['M',[x, y+ry]],
+					['C',[x, y+ry/num, x+rx/num, y, x+rx, y]],
+					['L',[x+w-rx, y]],
+					['C',[x+w-rx/num, y, x+w, y+ry/num, x+w, y+ry]],
+					['L',[x+w, y+h-ry]],
+					['C',[x+w, y+h-ry/num, x+w-rx/num, y+h, x+w-rx, y+h]],
+					['L',[x+rx, y+h]],
+					['C',[x+rx/num, y+h, x, y+h-ry/num, x, y+h-ry]],
+					['L',[x, y+ry]],
+					['Z',[]]
+				]);
+			}
+			break;
+		default:
+			break;
+	}
+
+	return d;
+
+};
+
+svgedit.utilities.addAttributesForConvertToPath = function( elem, attrs) {
+	// TODO: make this list global so that we can properly maintain it
+	// TODO: what about @transform, @clip-rule, @fill-rule, etc?
+	$.each(['marker-start', 'marker-end', 'marker-mid', 'filter', 'clip-path'], function() {
+		if (elem.getAttribute(this)) {
+			attrs[this] = elem.getAttribute(this);
+		}
+	});
+};
+
+// Function: getBBoxOfElementAsPath
+// Get the BBox of an element-as-path
 //
 // Parameters:
-// elem - Optional DOM element to get the angle for
+// elem - The DOM element to be probed
+// addSvgElementFromJson - Function to add the path element to the current layer. See canvas.addSvgElementFromJso
+// pathActions - If a transform exists, pathActions.resetOrientation() is used. See: canvas.pathActions.
```

Comment:
Ugh - this reminds me of a big TODO I had a long time ago where I wanted to pull all of canvas.pathActions into a separate module.  That 1000 lines of code shouldn't be sitting in svgcanvas.js :-/

(I'm not asking you to do any of that, it's just an FYI)